lib/renderer.js

```diff
@@ -1265,11 +1265,13 @@
             canvasDimensionsScale = 1,
             settingsPromise,
             resultPromise,
+            hasVectorMask = layer && layer.path && layer.path.enabled,
```

Comment:
At first glance, it seems like `layer.path.enabled` is too aggressive.  Are you sure that doesn't resolve to non-false for regular shape layers?